### DISCUSSION TITLE
JWT bearer token: Issued At (iat) claim is not present but seems like this is required by quarkus-oidc

### DISCUSSION THREAD

**Discussion by subashiniunny** [quarkusio/quarkus#31232] [Q&A]
Hello,
JWT Bearer tokens generated by our identity provider do not contain the Issued At (iat) claim. But it looks like quarkus-oidc expects this as I get the following error when trying to authenticate a bearer token:

> 2023-02-16 14:27:45,218 DEBUG [io.qua.oid.run.OidcProvider] (vert.x-eventloop-thread-5) Token verification has failed: No Issued At (iat) claim present.

Please let me know how I can work around this.
I was using Quarkus 2.4.2.Final but after seeing https://github.com/smallrye/smallrye-jwt/issues/469, I updated quarkus version to 2.9.2.Final and also tried setting the following, but still getting this error about required iat claim. What else am I missing?

> smallrye.jwt.time-to-live=-1

It looks like OidcProvider is not checking for a condition similar to the above. Can something similar be checked for before doing the following?
https://github.com/quarkusio/quarkus/blob/main/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java#L140

[a comment by an automated account is not shown]

**Reply by sberyozkin** [reply]
@subashiniunny For `quarkus-oidc`, `iat` must be present when it is dealing with the authorization code flow, but perhaps for bearer tokens this requirement can be relaxed, is there a reason you have to use `quarkus-oidc` instead of `quarkus-smallrye-jwt`  which is where setting `smallrye.jwt.time-to-live=-1` will be effective ?

**Reply by subashiniunny** [reply]
@sberyozkin 
It looks like `quarkus-smallrye-jwt` does not support calling the UserInfo endpoint. We need to fetch AD groups for a user by calling this endpoint. Hence I was trying to use `quarkus-oidc`. 
But due to the above issue, I might have to revert to `quarkus-smallrye-jwt` and have our own custom code to call the UserInfo endpoint. If this `iat` requirement can be relaxed in `quarkus-oidc` for bearer tokens, that would be great.
